sdk/tests/tutorials/marketdata/test_corporate_actions.py

```diff
@@ -1,49 +1,228 @@
 import unittest
+import uuid
+import pytz
+from datetime import datetime, timedelta
 
 import lusid
 import lusid.models as models
 from lusidfeature import lusid_feature
 from lusid.utilities.api_client_builder import ApiClientBuilder
 from utilities.credentials_source import CredentialsSource
 from utilities.test_data_utilities import TestDataUtilities
-import uuid
 
 
 class CorporateActions(unittest.TestCase):
-
     @classmethod
     def setUp(cls):
         # create a configured API client
-        api_client = ApiClientBuilder().build(CredentialsSource.secrets_path())
+        api_client = ApiClientBuilder().build(str(CredentialsSource.secrets_path()))
 
+        cls.instruments_api = lusid.InstrumentsApi(api_client)
+        cls.portfolios_api = lusid.PortfoliosApi(api_client)
+        cls.transaction_portfolios_api = lusid.TransactionPortfoliosApi(api_client)
         cls.corporate_actions_sources_api = lusid.CorporateActionSourcesApi(api_client)
 
     def get_guid(self):
         # creates random alphanumeric code
         return str(uuid.uuid4())[:12]
 
+    @lusid_feature("F12-4")
+    def test_name_change_corporate_action(self):
+        # Define details for the corporate action.
```

Comment:
Put a quick summary up here:

The code below shows processing a corporate action name change in LUSID:

1. Create two instruments (original and change)
2. Create a portfolio
3. Add transaction for original instrument
4. Create a corporate action source.
...